src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -166,32 +169,23 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
      * Called when a tag from footer is clicked
      * @param tag Tag clicked
      */
-    public onTagClicked = (tag: ITag): void => {
-        const selectedAsset = this.state.selectedAsset;
-        if (this.canvas.current.state.selectedRegions && this.canvas.current.state.selectedRegions.length) {
-            const selectedRegions = this.canvas.current.state.selectedRegions;
-            selectedRegions.map((region) => {
-                const selectedIndex = selectedAsset.regions.findIndex((r) => r.id === region.id);
-                const selectedRegion = selectedAsset.regions[selectedIndex];
-                const tagIndex = selectedRegion.tags.findIndex(
-                    (existingTag) => existingTag.name === tag.name);
-                if (tagIndex === -1) {
-                    selectedRegion.tags.push(tag);
-                } else {
-                    selectedRegion.tags.splice(tagIndex, 1);
-                }
-                if (selectedRegion.tags.length) {
-                    this.canvas.current.editor.RM.updateTagsById(selectedRegion.id,
-                        new TagsDescriptor(selectedRegion.tags.map((tempTag) => new Tag(tempTag.name,
-                            this.props.project.tags.find((t) => t.name === tempTag.name).color))));
-                } else {
-                    this.canvas.current.editor.RM.updateTagsById(selectedRegion.id, null);
-                }
-
-                return region;
-            });
-        }
-        this.onAssetMetadataChanged(selectedAsset);
+    public onTagClicked = (tag: ITag) => {
```

Comment:
Instead of having the `editor page` call into the `canvas` component to perform tag related operations I would like to see this model inverted where the `editor page` only sets up some internal state like `selectedTags` or similar which is then passed through props into the `canvas` where the `canvas` can then do the right thing based on its state.

Reply:
Let's discuss.

Reply:
Agreed. I'll re-work this and get back to you

Reply:
So the only thing in my mind that's keeping some of this in the canvas is the `selectedTags` issue. The `onTagClicked` needs to update all selected tags

Reply:
Oh wait, I think I know what to do here. Could help with the tag locking as well. I'll get back to you